deltacat/storage/interface.py

```diff
@@ -178,9 +183,10 @@ def download_delta(
     columns: Optional[List[str]] = None,
     file_reader_kwargs_provider: Optional[ReadKwargsProvider] = None,
     ray_options_provider: Callable[[int, Any], Dict[str, Any]] = None,
+    distributed_dataset_type: DistributedDatasetType = DistributedDatasetType.RAY_DATASET,
```

Comment:
I feel this overcomplicates the interface slightly. Rather than having a new parameter, could we not re-use `table_type` and `storage_type`? For example, if we want our deltas read into a Daft DataFrame, could we just specify something like `TableType.DAFT_DATAFRAME` and `StorageType.DISTRIBUTED` to achieve the same result?

Reply:
The reason why we needed a enums was, you can have RAY_DATASET with table type being pandas, arrow, or even numpy.